src/guake/guake_app.py

```diff
@@ -693,21 +696,27 @@ def set_final_window_rect(self):
         window_rect = screen.get_monitor_geometry(monitor)
 
         if os.environ.get('DESKTOP_SESSION') == "ubuntu":
-            unity_hide = self.client.get_int(KEY('/apps/compiz-1/plugins/'
-                                                 'unityshell/screen0/options/launcher_hide_mode'))
-            # launcher_hide_mode = 1 => autohide
-            if unity_hide != 1:
-                # Size of the icons for Unity in Ubuntu <= 12.04
-                # TODO Ubuntu 12.10 use dconf :
-                # /org/compiz/profiles/unity/plugins/unityshell/icon-size
+
+            # For Ubuntu 12.10 and above, use dconf :
+            # For Ubuntu 12.04 and below, use
+            # see if unity dock is hide => unity_hide
+            # and the width of unity dock. => unity_dock
+            if float(platform.dist()[1])>=12.10:
```

Comment:
Well at first you would want to check if you are on ubuntu or not, because ``float(21) > 12.10`` and that can be the case for many other OS.

But in general, I am not in favor of adding distro-specific pieces of code if we can avoid it.